features/FEATURE_BLE/targets/TARGET_CORDIO/CordioPalAttClient.h

```diff
@@ -318,11 +320,20 @@ class CordioAttClient : public ::ble::pal::AttClient {
      */
     static void att_client_handler(const attEvt_t* event)
     {
+        if (event->hdr.status == ATT_SUCCESS && event->hdr.event == ATT_MTU_UPDATE_IND) {
+            ble::pal::Gap::EventHandler *handler;
+            handler = ble::pal::vendor::cordio::Gap::get_gap().get_event_handler();
+            if (handler) {
+                handler->on_att_mtu_changed(event->hdr.param, event->mtu);
+            }
+            return;
+        }
+
         // all handlers are stored in a static array
         static const event_handler_t handlers[] = {
             &timeout_event_handler,
             &event_handler<ErrorResponseConverter>,
-            &event_handler<ExchangeMtuResponseConverter>,
+            //&event_handler<ExchangeMtuResponseConverter>,
```

Comment:
I think these handlers should only contain events that are allowed in existing transactions

Reply:
Gatt client is the only entity that can send an `Exchange MTU request` and receive an `Exchange MTU response`.
Similarly, a gatt server is the only entity that can receive an `Exchange MTU response` and send an `Exchange MTU response`.

Reply:
that's the thing, this event isn't what the name suggests, this is produced by either client or server when the size is negotiated (and only if it's different from existing one i.e. default)